src/frequenz/sdk/actor/power_distributing/_component_status/_ev_charger_status_tracker.py

```diff
@@ -0,0 +1,204 @@
+        )
+
+    def _is_stale(self, ev_data: EVChargerData) -> bool:
+        """Return whether the given data is stale."""
+        now = datetime.now(tz=timezone.utc)
+        stale = now - ev_data.timestamp > self._max_data_age
+        return stale
+
+    async def _run_forever(self) -> None:
+        """Run the status tracker forever."""
+        while True:
+            try:
+                await self._run()
+            except Exception:  # pylint: disable=broad-except
+                _logger.exception(
+                    "Restarting after exception in EVChargerStatusTracker.run()"
+                )
+                await asyncio.sleep(1.0)
+
+    def _handle_ev_data(self, ev_data: EVChargerData) -> ComponentStatusEnum:
+        """Handle new EV charger data."""
+        if self._is_stale(ev_data):
+            if self._last_status == ComponentStatusEnum.WORKING:
+                _logger.warning(
+                    "EV charger %s data is stale. Last timestamp: %s",
+                    self._component_id,
+                    ev_data.timestamp,
+                )
+            return ComponentStatusEnum.NOT_WORKING
+
+        if self._is_working(ev_data):
+            if self._last_status == ComponentStatusEnum.NOT_WORKING:
+                _logger.warning(
+                    "EV charger %s is in WORKING state.",
+                    self._component_id,
+                )
+            return ComponentStatusEnum.WORKING
+
+        if self._last_status == ComponentStatusEnum.WORKING:
+            _logger.warning(
+                "EV charger %s is in NOT_WORKING state. "
+                "Cable state: %s, component state: %s",
+                self._component_id,
+                ev_data.cable_state,
+                ev_data.component_state,
+            )
+        return ComponentStatusEnum.NOT_WORKING
+
+    def _handle_set_power_result(
+        self, set_power_result: SetPowerResult
+    ) -> ComponentStatusEnum:
+        """Handle a new set power result."""
+        if self._component_id in set_power_result.succeeded:
+            return ComponentStatusEnum.WORKING
+
+        self._blocking_status.block()
+        if self._last_status == ComponentStatusEnum.WORKING:
+            _logger.warning(
+                "EV charger %s is in UNCERTAIN state. Set power result: %s",
+                self._component_id,
+                set_power_result,
+            )
+        return ComponentStatusEnum.UNCERTAIN
+
+    async def _run(self) -> None:
+        """Run the status tracker."""
+        api_client = connection_manager.get().api_client
+        ev_data_rx = await api_client.ev_charger_data(self._component_id)
+        set_power_result_rx = self._set_power_result_receiver
+        missing_data_timer = Timer(self._max_data_age, SkipMissedAndDrift())
+
+        # Send initial status
+        await self._status_sender.send(
+            ComponentStatus(self._component_id, self._last_status)
```

Comment:
Hmm this means we will always send the default NOT_WORKING initially before correcting it no? Why not make default `None` and don't send any initial status, just let the loop update it?

Reply:
Worst case, the timer will fire and send the update

Reply:
Yes, but that would require some amount of refactoring, because the timer trigger would send something out only if the status has changed.  But I'm not sure that it should be changed.

This is an internal feature, and its values are used by the power distributor and the system bounds tracker.  So having it like this in practice would mean that PowerManager can report zero bounds right away on start up, until data is available.

Reply:
This is used as a hint to the power manager and *pool metrics whether a battery is usable or not.  We need to send something out right at the beginning, else they wait for a status to arrive, before streaming values out.

So changing this would mean changing all the users of this data, which is more than a few.

Sending the `NOT_WORKING` flag even though we don't know if it is working or not is I'm guessing the objection?  Maybe renaming it to `NOT_USABLE` would make it more general, and can be sent before we receive any data, or when missing data, or when the component is known to be not working?

Reply:
> else they wait for a status to arrive, before streaming values out.

I mean, optimally it (the status) would arrive right before the values, or do you mean they only request the values once they have the status?


I think the reason I am talking about this is, as a user, you see the logs like "Component changed to (NOT_)WORKING" etc and it can be worrying to see "NOT WORKING" at startup. I know it was for me :)
`NOT_USABLE` is not much better in that context, you simply don't expect a `NOT_` like that right at startup when things are actually working in order

Reply:
I think the way it is implemented, we see `EV charger 1008 status changed from ComponentStatusEnum.NOT_WORKING to ComponentStatusEnum.WORKING` almost immediately after the app starts.  And these are the first messages related to status that we see, almost immediately after app start:

```
WARNING:frequenz.sdk.actor.power_distributing._component_status._ev_charger_status_tracker:EV charger 1008 is in WORKING state.
INFO:frequenz.sdk.actor.power_distributing._component_status._ev_charger_status_tracker:EV charger 1008 status changed from ComponentStatusEnum.NOT_WORKING to ComponentStatusEnum.WORKING
WARNING:frequenz.sdk.actor.power_distributing._component_status._ev_charger_status_tracker:EV charger 1007 is in WORKING state.
INFO:frequenz.sdk.actor.power_distributing._component_status._ev_charger_status_tracker:EV charger 1007 status changed from ComponentStatusEnum.NOT_WORKING to ComponentStatusEnum.WORKING
```

If the BatteryPool says "Changed to NOT_WORKING", we'll have to change that separately.

Reply:
In the battery pool too, it doesn't seem to say "NOT_WORKING" at startup, just:

```
INFO:frequenz.sdk.actor.power_distributing._component_status._battery_status_tracker:battery 1001 changed status ComponentStatusEnum.WORKING
INFO:frequenz.sdk.actor.power_distributing._component_status._battery_status_tracker:battery 1004 changed status ComponentStatusEnum.WORKING

Reply:
Without having seen the changes yet, and not being very familiar with how this is used (so high risk of saying nonsense), how about `INITIALIZING` instead of `UNKNOWN`, if we can use a special state for this initialization step?

Reply:
I guess related to this, reading the commit message:

> It reports an EV charger as `WORKING` when an EV is connected to it,
> and power can be allocated to it, and `NOT_WORKING` otherwise.

I guess this status comes from batteries, right? Maybe it would be better to rename `WORKING`/`NOT_WORKING` to `USABLE`/`UNUSABLE` or `AVAILABLE`/`UNAVAILABLE`, it looks a bit misleading to say a EV charger is "not working" just because an EV is not connected to it.

Reply:
It could also be because of EV charger in error state.  Then someone might have to go and reset.

This used for telling the Power Distributor whether power can be allocated to a component or not.  And we need alternatives for WORKING, NOT_WORKING, UNCERTAIN that would be suitable for batteries, EV chargers, PV inverters, CHP, etc.  

Maybe USABLE/NOT_USABLE is the way to go.  UNUSABLE is not distinct enough for my eyes, I might mix things up, especially when reading.

Reply:
But would keep for a separate PR: https://github.com/frequenz-floss/frequenz-sdk-python/issues/903